lib/matplotlib/axes/_axes.py

```diff
@@ -6142,6 +6147,16 @@ def _normalize_input(inp, ename='input'):
                 raise ValueError(
                     'weights should have the same shape as x')
 
+        # Combine the masks from x[i] and w[i] (if applicable) into a single
+        # mask and apply it to both.
+        if not input_empty:
+            for i in range(len(x)):
+                mask_i = x[i].mask
+                if w[i] is not None:
+                    mask_i = mask_i | w[i].mask
```

Comment:
Ahh, that's pretty nice, I didn't read the docs carefully enough to see the part about any points masked in one arg getting masked and deleted in the others.  I'm not sure I understand what ['All input arguments that are not passed unchanged'](http://matplotlib.org/api/cbook_api.html#matplotlib.cbook.delete_masked_points) means though...